### PR TITLE
[WIP] Feature/refactor

### PR DESCRIPTION
**Pull request by nmsohn**
# **CLOSED: THIS ONE IS MERGED WITH `3-add-cases-to-folders` BRANCH**

## CHANGELOG
- util added
- folder/cases refactored
- comments removed
- mocha unit test for folder added
- folder, case migrations updated
- helmet added

## NOTE
I temporarily disabled auth function in controllers
At the moment you need to set userId = 0;

### CODE REVIEW

File: app/controller/folders.js [hunk 1]
```diff
@@ -1,83 +1,86 @@
...
-  Folders.insert(data, userId, err => {
-    if (err) {
-      res.status(500).send(err.stack)
-    } else {
-      res.status(201).send('Created folder')
-    }
-  })
+export function create(req, res) {
+	const data = Object.keys(req.body) ? res.json(req.body) : null;
+	const userId = req.user.sub || 0;
```
Comment (nmsohn) [inline app/controller/folders.js]: TypeError: Cannot read property 'sub' of undefined. There is no user object in the request @tehnools

File: app/controller/folders.js [hunk 2]
```diff
@@ -1,83 +1,86 @@
-const Folders = require('../model/folders')
+const Folders = require("../model/folders");
 
-// TODO: Refactoring
-export function create (req, res) {
-  const data = req.body
-  const userId = req.user.sub || 0
-
-  // Auth Check, to be replaced by authorization middlewares
-  // if (req.user['https://openlaw.nz/roles'] !== 'admin') {
```
Comment (nmsohn) [inline app/controller/folders.js]: https://openlaw.nz/role is not found

File: app/controller/folders.js [hunk 3]
```diff
@@ -1,83 +1,86 @@
...
 
-  Folders.selectOne(data, userId, folderId, err => {
-    if (err) {
-      res.status(500).send(err.stack)
-    } else {
-      res.status(201).send('Read folder')
-    }
-  })
+	Folders.selectOne(data, userId, folderId, (err) => {
+		if (err) {
```
Comment (nmsohn) [inline app/controller/folders.js]: do we need to pass err?

File: app/controller/folders.js (hunk 1 above)
Comment (tehnools) [reply app/controller/folders.js]: you have to have auth0 middlewares. I'll send the password and username now

File: app/controller/folders.js (hunk 3 above)
Comment (tehnools) [reply app/controller/folders.js]: we need to pass errors in res.status(error code).send(err message) i have created a solution for that for the code i am creating now.

**Pull request closed without merging**